provider-service/kfp/internal/event_flow.go

```diff
@@ -0,0 +1,245 @@
+}
+
+func (kef *KfpEventFlow) toRunCompletionEventData(message StreamMessage[*unstructured.Unstructured]) (StreamMessage[*common.RunCompletionEventData], error) {
+	runCompletionEventData, err := kef.eventForWorkflow(kef.context, message.Message)
+	if err != nil {
+		message.OnFailureHandler()
+		return StreamMessage[*common.RunCompletionEventData]{}, err
+	}
+	return StreamMessage[*common.RunCompletionEventData]{
+		Message:            runCompletionEventData,
+		OnCompleteHandlers: message.OnCompleteHandlers,
+	}, nil
+}
+
+func (kef *KfpEventFlow) eventForWorkflow(ctx context.Context, workflow *unstructured.Unstructured) (*common.RunCompletionEventData, error) {
+	status, hasFinished := runCompletionStatus(workflow)
+	if !hasFinished {
+		kef.Logger.V(2).Info("ignoring workflow that hasn't finished yet")
+		return nil, nil
+	}
+
+	workflowName := workflow.GetName()
+
+	modelArtifacts, err := kef.MetadataStore.GetServingModelArtifact(ctx, workflowName)
+	if err != nil {
+		kef.Logger.Error(err, "failed to retrieve serving model artifact")
+		return nil, err
+	}
+
+	runId := workflow.GetLabels()[pipelineRunIdLabel]
+	resourceReferences, err := kef.KfpApi.GetResourceReferences(ctx, runId)
+	if err != nil {
+		kef.Logger.Error(err, "failed to retrieve resource references")
+		return nil, err
+	}
+
+	// For compatability with resources created with v0.3.0 and older
+	if resourceReferences.PipelineName.Empty() {
+		pipelineName := getPipelineName(workflow)
+		if pipelineName == "" {
+			kef.Logger.Info("failed to get pipeline name from workflow")
+			return nil, nil
+		}
+
+		resourceReferences.PipelineName.Name = pipelineName
+	}
+
+	return &common.RunCompletionEventData{
+		Status:                status,
+		PipelineName:          resourceReferences.PipelineName,
+		RunConfigurationName:  resourceReferences.RunConfigurationName.NonEmptyPtr(),
+		RunName:               resourceReferences.RunName.NonEmptyPtr(),
+		RunId:                 runId,
+		ServingModelArtifacts: modelArtifacts,
+		PipelineComponents:    nil,
+		Provider:              kef.ProviderConfig.Name,
+	}, nil
+}
+
+func runCompletionStatus(workflow *unstructured.Unstructured) (common.RunCompletionStatus, bool) {
+	switch workflow.GetLabels()[workflowPhaseLabel] {
+	case string(argo.WorkflowSucceeded):
+		return common.RunCompletionStatuses.Succeeded, true
+	case string(argo.WorkflowFailed), string(argo.WorkflowError):
+		return common.RunCompletionStatuses.Failed, true
+	default:
+		return "", false
+	}
+}
+
+func getPipelineNameFromAnnotation(workflow *unstructured.Unstructured) string {
+	specString := workflow.GetAnnotations()[pipelineSpecAnnotationName]
+	spec := &PipelineSpec{}
+	if err := json.Unmarshal([]byte(specString), spec); err != nil {
+		return ""
+	}
+
+	return spec.Name
+}
+
+func getPipelineName(workflow *unstructured.Unstructured) (name string) {
+	name = getPipelineNameFromAnnotation(workflow)
+
+	if name == "" {
+		name = getPipelineNameFromEntrypoint(workflow)
+	}
```

Comment:
```suggestion
	if name = getPipelineNameFromEntrypoint(workflow); name == "" {
		name = getPipelineNameFromEntrypoint(workflow)
	}
```

Reply:
changed in 847762891ac9e4c53e697be2fbd16d03bfb70260